src/main/java/com/synopsys/integration/jenkins/scan/SecurityScanner.java

```diff
@@ -1,65 +1,71 @@
+        this.run = run;
         this.listener = listener;
         this.launcher = launcher;
         this.workspace = workspace;
         this.envVars = envVars;
         this.scannerArgumentService = scannerArgumentService;
     }
 
-    public int runScanner(Map<String, Object> scanParameters) {
-        BlackDuckParametersService blackDuckParametersService = new BlackDuckParametersService();
+    public int runScanner(Map<String, Object> scanParams) throws ScannerJenkinsException {
+        BlackDuckParametersService blackDuckParametersService = new BlackDuckParametersService(listener);
 
-        BridgeDownloadParameters bridgeDownloadParameters = new BridgeDownloadParameters();
-        BridgeDownloadParametersService bridgeDownloadParametersService = new BridgeDownloadParametersService();
-        BridgeDownloadParameters bridgeDownloadParams = bridgeDownloadParametersService.getBridgeDownloadParams(scanParameters, bridgeDownloadParameters);
+        BridgeDownloadParameters bridgeDownloadParameters = new BridgeDownloadParameters(workspace, listener);
+        BridgeDownloadParametersService bridgeDownloadParametersService = new BridgeDownloadParametersService(workspace, listener);
+        BridgeDownloadParameters bridgeDownloadParams = bridgeDownloadParametersService.getBridgeDownloadParams(scanParams, bridgeDownloadParameters);
 
-        Map<String, Object> blackDuckParameters = blackDuckParametersService.prepareBlackDuckParameterValidation(scanParameters);
+        Map<String, Object> blackDuckParameters = blackDuckParametersService.prepareBlackDuckParameterValidation(scanParams);
         int scanner = 1;
 
         if (blackDuckParametersService.performBlackDuckParameterValidation(blackDuckParameters)
                 && bridgeDownloadParametersService.performBridgeDownloadParameterValidation(bridgeDownloadParams)) {
-
-            FilePath bridgeInstallationPath = new FilePath(new File(bridgeDownloadParams.getBridgeInstallationPath()));
-            List<String> commandLineArgs = scannerArgumentService.getCommandLineArgs(scanParameters);
-
-            BridgeDownloadManager bridgeDownloadManager = new BridgeDownloadManager();
+            BridgeDownloadManager bridgeDownloadManager = new BridgeDownloadManager(workspace, listener);
             boolean isBridgeDownloadRequired = bridgeDownloadManager.isSynopsysBridgeDownloadRequired(bridgeDownloadParams);
+
             if (isBridgeDownloadRequired) {
                 initiateBridgeDownloadAndUnzip(bridgeDownloadParams);
+            } else {
+                listener.getLogger().printf(LogMessages.BRIDGE_INSTALLATION_ALREADY_FOUND_IN_PATH, bridgeDownloadParams.getBridgeInstallationPath());
             }
 
-            Utility.copyRepository(workspace.getRemote(), bridgeDownloadParams.getBridgeInstallationPath());
-
-            printMessages(LogMessages.START_SCANNER);
+            Utility.copyRepository(bridgeDownloadParams.getBridgeInstallationPath(), workspace, listener);
```

Comment:
If we are copying source code to synopsys bridge path, we should not copy the source code to synopsys bridge path. Instead, we can copy the source code to workspace and from workspace context we can refer to synopsys-bridge.

Reply:
Removed this logic as discussed. Now we are downloading and unzipping bridge in default path/user specified path and invoking `synopsys-bridge` from workspace context (source code is also in workspace context)  [updates can be found here - https://github.com/synopsys-sig/synopsys-security-scan-plugin/blob/dev/src/main/java/com/synopsys/integration/jenkins/scan/service/ScannerArgumentService.java#L107]

Note - since now we are invoking bridge form workspace context so the diagnostics are generating in workspace.
Therefore we have updated the diagnostics archiving path. [https://github.com/synopsys-sig/synopsys-security-scan-plugin/blob/dev/src/main/java/com/synopsys/integration/jenkins/scan/SecurityScanner.java#L85]